cmd/flux/bootstrap_gitlab.go

```diff
@@ -133,6 +143,11 @@ func bootstrapGitLabCmdRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("--token-auth and --deploy-token-auth cannot be set both.")
 	}
 
+	if !gitlabArgs.private {
```

Comment:
@darkowlzz I added the message in https://github.com/fluxcd/flux2/pull/4866/files#diff-133675ca6f2909722d0f3d6874fc348dddf836f06748df653ef821218be4362cR148 to avoid the logic of checking. Adding `--private=false` will be a user action and it will always conflict with `--visibility=private` because the latter is the default value for visibility. We can't check for this case, so I considered it to be a more consistent experience if we just warn the user when they set private false.